subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/CachingModuleComponentRepository.java

```diff
@@ -201,8 +201,10 @@ private void resolveComponentMetaDataFromCache(ModuleComponentIdentifier moduleC
                 }
                 LOGGER.debug("Detected non-existence of module '{}' in resolver cache '{}'", moduleComponentIdentifier, delegate.getName());
                 result.missing();
-                // When age == 0, verified since the start of this build, assume still missing
-                result.setAuthoritative(cachedMetadata.getAgeMillis() == 0);
+                if (result.isRetryMissing()) {
+                    // When age == 0, verified since the start of this build, assume still missing
+                    result.setAuthoritative(cachedMetadata.getAgeMillis() == 0);
+                }
```

Comment:
:x: If we handle `retryMissing` in the caller, this change can be reverted. (refer to other comments)